pages/security/message-security/openpgp/gpg-best-practices/en.text

```diff
@@ -131,11 +131,21 @@ pre. gpg --output revoke.asc --gen-revoke '<fingerprint>'
 
-In this scenario, your primary key is used only for certifications, which happen infrequently.
+If you are using public-key encryption with GnuPG, you [[already have a separate encryption subkey -> http://security.stackexchange.com/questions/43590/pgp-why-have-separate-encryption-subkey]] that was automatically created when you generated your keys. However, you also create an additional subkey for signing data/messages, and then securely store your primary key offline (perhaps on removable media or a separate computer that isn't connected to the internet). The things you do most frequently with GPG -- signing and encrypting messages/data -- are done using your subkeys, which stay with you on your personal computer for everyday use. However, your primary key (which is not on the same computer) is now only used in a few special circumstances (which generally happen relatively infrequently): 
+
+(1) certifying other people's keys, 
+(2) creating new subkeys, 
+(3) editing/revoking subkeys. 
+
+To understand the benefit of all of this, consider what would happen if you were using the default GnuPG setup (a primary signing/certification key and an encryption subkey) on your laptop, and that your laptop was then stolen by an attacker. Let's say that you had been smart enough to encrypt your hard disk, picked a very strong passphrase for your key, and had generated a revocation certificate and set an expiration date. So it's unlikely that an attacker will be able to decrypt your messages/data, because even if they decrypt your disk and get access to your private keys, they still have to find your passphrase. Furthermore, you can revoke your keys with the revocation certificate so everyone knows not to trust them anymore (and they'll expire anyway). However, you still have a major problem: over the years, you've been building up [[trust -> https://en.wikipedia.org/wiki/Web_of_trust]] in the form of other PGP users signing your primary key. You now have to start from scratch rebuilding your web of trust by getting everyone to re-sign your new key, and you've lowered the trust of all the keys you had signed with your old primary key. It can be very troublesome and time consuming to go through this process again, and its disruptive of the web of trust ...
+
+The process described above, however, solves this problem! Let's imagine instead that your stolen laptop only contained your encryption subkey and your signing subkey, and that your primary key was safely stored somewhere else. This time you don't have to revoke your primary key, which is the one that everyone else has been certifying -- i.e. you don't have to rebuild your web of trust. All you have to do now is revoke your subkeys and generate new ones, using your primary key. 
+
+The process to generate subkeys is somewhat involved, but there are several helpful guides that have been written elsewhere describing how to do it. You can find them with a cursory internet search for "gpg subkeys" ...
```

Comment:
for me the last sentence here is the key problem with this change. i really appreciate the documentation of the "why", but stopping short of doing the "how" defeats the whole purpose here, especially if we just throw users at a search engine instead of a direct link (did you mean [this page](https://oguya.ch/posts/2016-04-01-gpg-subkeys/)? if so, just link to it already :))...